src/components/Identity/setCustomerIds.js

```diff
@@ -1,10 +1,9 @@
-import { assign } from "../../utils";
 import createEvent from "../DataCollector/createEvent";
 import { validateCustomerIds } from "./util";
 import processCustomerIds from "./processCustomerIds";
 
-const makeServerCall = (payload, lifecycle, network) => {
-  return lifecycle.onBeforeDataCollection(payload).then(() => {
+const makeServerCall = (payload, processedIds, lifecycle, network) => {
```

Comment:
I agree with Joe's comment above about not passing processedIds into the onBeforeDataCollection.  Could you add the customerIds to the payload here?

Reply:
I could, do you suggest I add from here and use payload.toJSON to get customerIds inside the onbeforeHooke?

Reply:
Why do you need the customerIds in the onBeforeHook?